frontend/src/App.tsx

```diff
@@ -1270,6 +1303,25 @@ export class App extends PureComponent<Props, State> {
     this.openDialog(newDialog)
   }
 
+  inIFrame = (): boolean => {
```

Comment:
Shouldn't this function be moved to `utils.ts`, next to `isEmbeddedInIFrame` and `isInChildFrame`? BTW: maybe one of these functions already does what you're trying to check? Also, why this function can throw?

Reply:
Checked this out and seems like `isInChildFrame` is better than `isEmbeddedInIFrame`, so now `isEmbeddedInIFrame` is removed and I use `isInChildFrame` instead. Thank you 🙇